### PR TITLE
=act,build #23719 multi-release-jar akka.util.Unsafe for JDK9

### LINKED ISSUE #23719: Separate a multi-release-jar akka.util.Unsafe for JDK9
In there we should encapsulate our use of unsafe with "safe" or alternative impls that should be used when JDK9 is used.

### PR DESCRIPTION
**Pull request by ktoso**
Will resolve #23719 but needs sbt 1.0 build, so I may retarget this PR I guess.

### CODE REVIEW

File: akka-actor/src/main/java-jdk9/akka/util/Unsafe.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+package akka.util;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+
+/**
+ * INTERNAL API
+ */
+public final class Unsafe {
```
Comment (ktoso) [inline akka-actor/src/main/java-jdk9/akka/util/Unsafe.java]: Will add more comments here, but notice that this file can assume being JDK9

`akka-actor/src/main/java-jdk9/akka/util/Unsafe.java`

File: akka-actor/src/main/scala/akka/util/Unsafe.java [hunk 2]
```diff
@@ -13,7 +12,6 @@
     public static final sun.misc.Unsafe instance;
 
     private static final long stringValueFieldOffset;
-    private static final boolean isJavaVersion9Plus;
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/Unsafe.java]: this file can assume being JDK8-

**Comment by ktoso** [comment]
cc @guidomedina this is what I meant in the Unsafe PR when I said "there won't be a need for the `jf(jdk9)` code", we can avoid that via mr-jars, see more: https://github.com/sbt/sbt-multi-release-jar (we developed the plugin, the feature is a JDK9 thing though of course)

**Comment by ktoso** [comment]
Failure is the expected formatting thingy.

**Comment by ktoso** [comment]
Merging into the `build-jdk9` branch so we have all things ready in there, will verify locally once more then.

**Comment by ktoso** [comment]
Please review though!

**Review by patriknw — COMMENTED** [review]
code is LGTM

**Comment by guidomedina** [comment]
Will this have the same effect as building Akka for Scala `2.11` and `2.12` in that a jar is released for Java 8 and another for Java 9?

**Comment by ktoso** [comment]
No. Please see https://github.com/sbt/sbt-multi-release-jar and http://openjdk.java.net/jeps/238 for details.

The class ships always, and is only loaded by JDK9 instead of the "normal one".

**Comment by guidomedina** [comment]
Oh no, I believe you and is great, that's really great I like it !!!

**Pull request merged**